### PR TITLE
feat: send templated email when COJ received

### PR DESCRIPTION
**Pull request by Judge40**
When a COJ is received an event will be emitted containing the details of the programme membership and signed COJ.
Listen for COJ received events and send a personalised email to notification the user that their COJ was received.

The email should address `Dr <Surname>` when possible, otherwise `Dear Doctor` should be used.
If available, the LO and received date should be included in the email. A link should be provided to give easy access to TSS for download of PDF copies - it is not possible to route them directly to the COJ view due to application state, so they will be sent to the programmes tab instead.

TIS21-5058
TIS21-5055

### CODE REVIEW

File: src/main/java/uk/nhs/tis/trainee/notifications/event/ConditionsOfJoiningListener.java [hunk 1]
```diff
@@ -0,0 +1,105 @@
...
+      throws MessagingException {
+    log.info("Handling COJ received event {}.", event);
+    Map<String, Object> templateVariables = new HashMap<>();
+    templateVariables.put("domain", appDomain);
+    templateVariables.put("managingDeanery", event.managingDeanery());
+
+    // Convert the UTC timestamp to the receiver's timezone.
+    if (event.conditionsOfJoining() != null && event.conditionsOfJoining().syncedAt() != null) {
+      ZonedDateTime syncedAt = ZonedDateTime.ofInstant(event.conditionsOfJoining().syncedAt(),
+          ZoneId.of(timezone));
```
Comment (ReubenRobertsHEE) [inline src/main/java/uk/nhs/tis/trainee/notifications/event/ConditionsOfJoiningListener.java:83]: 🌮

File: src/main/resources/application.yml [hunk 2]
```diff
@@ -4,7 +4,14 @@ server:
     context-path: /notifications
 
 application:
+  domain: ${APP_DOMAIN:}
+  email:
+      sender: ${EMAIL_SENDER}
```
Comment (ReubenRobertsHEE) [inline src/main/resources/application.yml:9]: I assume bounce-backs will go to this mailbox - will we have access to this? (And/or the ability to configure an alarm if it has mail, that would be cool!)

Comment (Judge40) [reply src/main/resources/application.yml:9]: Bounces and complaints can be routed to SNS/SQS, then we can have either a simple alarm on the SQS queue or something sat at the end of that queue to automate the follow-up.

**Review by ReubenRobertsHEE — APPROVED** [review]
Looks good.

**Pull request merged**